fineract-provider/src/main/java/org/apache/fineract/infrastructure/creditbureau/service/CreditReportWritePlatformServiceImpl.java

```diff
@@ -146,20 +140,24 @@ public String addCreditReport(Long bureauId, File creditReport, FormDataContentD
 
     }
 
-    private Optional<String> getCreditBureau(Long creditBureauID) {
+    private String getCreditBureau(Long creditBureauID) {
 
         if (creditBureauID != null) {
             Optional<CreditBureau> creditBureau = this.creditBureauRepository.findById(creditBureauID);
 
             if (creditBureau.isEmpty()) {
-                return Optional.empty();
+                baseDataValidator.reset().failWithCode("creditBureau.has.not.been.Integrated");
```

Comment:
You could swap the condition and get rid of this exceptional case.
So it could be 
```
if (!creditBureau.isEmpty()) {
    return creditBureau.get().getName()
}
```

And the later exception throw logic will catch the error cases anyway.